lib/plugins/leetcode.js

```diff
@@ -538,19 +538,118 @@ plugin.login = function(user, cb) {
+  const leetcodeUrl = config.sys.urls.github_login;
+  const _request = request.defaults({jar: true});
+  _request('https://github.com/login', function(e, resp, body) {
+    const authenticityToken = body.match(/name="authenticity_token" value="(.*?)"/);
+    if (authenticityToken === null) {
+      return cb('Get GitHub token failed');
+    }
+    const options = {
+      url:     'https://github.com/session',
+      method:  'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      followAllRedirects: true,
+      form:               {
+        'login':              user.login,
+        'password':           user.pass,
+        'authenticity_token': authenticityToken[1],
+        'utf8':               encodeURIComponent('✓'),
+        'commit':             encodeURIComponent('Sign in')
+      },
+    };
+    _request(options, function(e, resp, body) {
+      if (resp.statusCode !== 200) {
+        return cb('GitHub login failed');
+      }
+      _request.get({url: leetcodeUrl}, function(e, resp, body) {
+        const redirectUri = resp.request.uri.href;
+        if (redirectUri !== 'https://leetcode.com/') {
+          return cb('GitHub login failed or GitHub did not link to leetcode');
+        }
+        const cookieData = parseCookie(resp.request.headers.cookie, cb);
+        saveAndGetUser(user, cb, cookieData);
+      });
+    });
+  });
+};
+
+plugin.linkinedLogin = function(user, cb) {
+  const leetcodeUrl = config.sys.urls.linkined_login;
+  const _request = request.defaults({
+    jar:     true,
+    headers: {
+      'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/73.0.3683.86 Safari/537.36'
+    }
+  });
+  _request('https://www.linkedin.com', function(e, resp, body) {
+    if ( resp.statusCode !== 200) {
+      return cb('Get linkedin session failed');
+    }
+    const authenticityToken = body.match(/input name="loginCsrfParam" value="(.*)" /);
+    if (authenticityToken === null) {
+      return cb('Get Linkedin token failed');
+    }
+    const options = {
+      url:     'https://www.linkedin.com/uas/login-submit',
+      method:  'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      followAllRedirects: true,
+      form:               {
+        'session_key':      user.login,
+        'session_password': user.pass,
+        'loginCsrfParam':   authenticityToken[1],
+        'trk':              'guest_homepage-basic_sign-in-submit'
+      },
+    };
+    _request(options, function(e, resp, body) {
+      if (resp.statusCode !== 200) {
+        return cb('Linkedin login failed');
+      }
+      _request.get({url: leetcodeUrl}, function(e, resp, body) {
+        const redirectUri = resp.request.uri.href;
+        if (redirectUri !== 'https://leetcode.com/') {
+          return cb('Linkedin login failed or Linkedin did not link to leetcode');
```

Comment:
LinkedIn

LeetCode

Reply:
Sorry for that, I will correct them~
Thank you very much for your help ~